tendermint/Cargo.toml

```diff
@@ -33,36 +33,66 @@ rustdoc-args = ["--cfg", "docsrs"]
 crate-type = ["cdylib", "rlib"]
 
 [dependencies]
-anomaly = "0.2"
+anyhow = {version = "1.0", default-features = false}
```

Comment:
I try `eyre` and pare with `displaydoc`, I meet with compiler  error. so I try `anyhow`, this anyhow crate handle error crate have huge number statrs.  Your have any advice.

Reply:
I don't think it's necessarily bad to use `anyhow`, although I think it would be good to discuss the tradeoffs in #669 and be clear about why it's the right tool for the job and be sure about the overall error handling strategy.

I'll note `anyhow` is generally used for "top level" error handling in applications (which applies to `eyre` as well, and `anomaly` as it's currently being used in this crate), whereas at the library level something like `thiserror` or `displaydoc` are a better fit as they present concrete error types (often an enum over error cases), which makes it easier to handle and respond to errors in library-level code.

Reply:
> I'll note anyhow is generally used for "top level" error handling in applications (which applies to eyre as well, and anomaly as it's currently being used in this crate), whereas at the library level something like thiserror or displaydoc are a better fit as they present concrete error types (often an enum over error cases), which makes it easier to handle and respond to errors in library-level code.

Agreed. I would rather have the `tendermint` and `ibc` crate expose concrete error types rather than boxed errors with `anomaly` or `eyre` (wether or not the latter eventually gets no-std support).

It's a lot of work though, and not directly related to this PR since we were already using `anomaly` in both `tendermint` and `ibc`, so I don't mind if we stick with `anyhow` for now if it makes implementing no-std support easier. We can remove the dependency on `anomaly` at a later stage.

@soareschen What do you think?

Reply:
Here is the documentation for `flex-error`: https://docs.rs/flex-error/0.1.1/flex_error/

Reply:
Ok, thx.

Reply:
@soareschen `error[E0271]: type mismatch resolving `<ProposerPriority as TryFrom<i64>>::Error == ErrorReport<TendermintErrorDetail, eyre::Report>`
   --> tendermint/src/validator.rs:177:56
    |
177 |             proposer_priority: value.proposer_priority.try_into()?,
    |                                                        ^^^^^^^^ expected enum `Infallible`, found struct `ErrorReport`
    |
    = note: expected enum `Infallible`
             found struct `ErrorReport<TendermintErrorDetail, eyre::Report>`

error[E0271]: type mismatch resolving `<Time as TryFrom<tendermint_proto::google::protobuf::Timestamp>>::Error == ErrorReport<TendermintErrorDetail, eyre::Report>`
  --> tendermint/src/vote.rs:83:40
   |
83 |             timestamp: value.timestamp.map(TryInto::try_into).transpose()?,
   |                                        ^^^ expected enum `Infallible`, found struct `ErrorReport`
   |
   = note: expected enum `Infallible`
            found struct `ErrorReport<TendermintErrorDetail, eyre::Report>``
How to deal with the error? Can you give me advice? And I try switch anyhow trace smelly different, I add cargo.toml feature = anyhow_tracer, However, It is eyre.

Reply:
@romac @tony-iqlusion @soareschen I want to check if the error handling library has been switched to `flex-error`.

Reply:
It is not yet switched. Right now I am just prototyping on the feasibility. Sorry for the confusion. For the latest version, to use the anyhow tracer, you will have to disable default features and enable only `anyhow_tracer`:

```
flex-error = { version = "0.1.1", default-features = false, features = ["anyhow_tracer"] }
```

The development is still ongoing, so you might want to hold back from using `flex-error` for now.

Reply:
@soareschen @romac https://github.com/informalsystems/tendermint-rs/pull/890 , This is my work result. switch flex-error.